pkg/operator/configobserver/cloudprovider/observe_cloud_volume_plugin.go

```diff
@@ -0,0 +1,96 @@
+package cloudprovider
+
+import (
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/cloudprovider"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+// NewCloudVolumePluginObserver returns a new cloudprovider observer for syncing cloud provider specific
+// information to controller-manager, leaving only storage controller loops enabled.
+func NewCloudVolumePluginObserver(targetNamespaceName string, cloudVolumePluginPath, cloudProviderConfigPath []string) configobserver.ObserveConfigFunc {
+	cloudObserver := &cloudProviderObserver{
+		targetNamespaceName:     targetNamespaceName,
+		cloudProviderNamePath:   cloudVolumePluginPath,
+		cloudProviderConfigPath: cloudProviderConfigPath,
+	}
+	return cloudObserver.ObserveCloudVolumePlugin
+}
+
+// ObserveCloudVolumePlugin fills in the cluster-name extended argument for the controller-manager with the cluster's infra ID
+func (c *cloudProviderObserver) ObserveCloudVolumePlugin(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (ret map[string]interface{}, errs []error) {
+	defer func() {
+		ret = configobserver.Pruned(ret, c.cloudProviderConfigPath, c.cloudProviderNamePath)
+	}()
+
+	listers := genericListers.(InfrastructureLister)
```

Comment:
What happens if this assertion fails?

Reply:
It should not with proper use. The consumer of this method will be a part of the list [here](https://github.com/openshift/cluster-kube-controller-manager-operator/blob/35aa6d524d9707699630f04565d7250ccf9118a0/pkg/operator/configobservation/configobservercontroller/observe_config_controller.go#L82), this argument is populated internally. It is done similarly in other places.